runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -119,6 +120,20 @@ private void waitForWorkflow(@NonNull UUID workflow, @NonNull ManagementClient c
         throw new TimeoutException();
     }
 
+    /*
+     * private method to handle exceptions while sendRequest()
+     */
+    private void handleException(RuntimeException runtimeException) {
+        final Throwable cause = ExceptionUtils.getRootCause(runtimeException);
+        if (cause instanceof TimeoutException ||
+                cause instanceof IllegalStateException ||
+                cause instanceof NetworkException) {
```

Comment:
This seems incorrect. IllegalStateException and NetworkException are RuntimeException, so the cause returned by ExceptionUtils.getRootCause() is null, so "instance of" will fail, and exception will be thrown without being retried.

Here is a quick example for reference:
```java
@Test
    public void xxx() {
        try {
            throw new NetworkException("s", "t");
        } catch (RuntimeException e) {
            final Throwable cause = ExceptionUtils.getRootCause(e);
            if (cause instanceof NetworkException) {
                System.out.println(1111);
            } else {
                System.out.println(2222);
            }
        }
    }
```
The code prints 222 because `cause` is actually null.